projects/unipilot/index.js

```diff
@@ -0,0 +1,136 @@
+const sdk = require("@defillama/sdk");
+const { request, gql } = require("graphql-request");
+
+const { getChainTransform } = require("../helper/portedTokens");
+const getPositionDetails = require("./abis/getPositionDetails.json");
+
+const GRAPH_URL = {
+  ethereum:
+    "https://api.thegraph.com/subgraphs/name/unipilotvoirstudio/stats-v2",
+};
+
+const vaultsQuery = gql`
+  {
+    vaults(first: 1000) {
+      id
+      token0 {
+        id
+      }
+      token1 {
+        id
+      }
+    }
+  }
+`;
+
+function protocolTvl(chain) {
+  return async (timestamp, block, chainBlocks) => {
+    const balances = {};
+
+    const res = await request(GRAPH_URL[chain], vaultsQuery);
```

Comment:
hi, is there a way to fetch the vault list from on chain calls? or maybe by reading events?

Reply:
@BilalMir135 I made the change here, does this look right to you? 
https://github.com/DefiLlama/DefiLlama-Adapters/pull/2961